src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -550,6 +576,35 @@ def batchComplete(self, result):
         msg = "Fitting completed successfully in: %s s.\n" % GuiUtils.formatNumber(elapsed)
         self.parent.communicate.statusBarUpdateSignal.emit(msg)
 
+    def outputConstraintError(self, exception):
+        """
+        Returns an error string highlighting the error in the constraint.
+        The argument is an exception that was raised by expression.py, which are interpreted as following:
+        CyclicDefinitionError: parameter has cyclic dependencies
+        UnknownParameterNameError: unknown symbol found in constraint
+        ConstraintSyntaxError: syntax problem such as missing parentheses
+        """
+        assert(isinstance(exception, Exception))
+        if isinstance(exception, CyclicDefinitionError):
+            error_message = "Cyclic dependency with parameter: <b>%s</b>"%(exception.parameter)
+
+        elif isinstance(exception, UnknownParameterNameError):
+            constraint = exception.constraint
+            for symbol in exception.unknown_symbol:
+                constraint = constraint.replace(symbol, "<b>%s</b>" % (symbol))
+            error_message = "Unknown symbol: %s" % (constraint)
+
+        elif isinstance(exception, ConstraintSyntaxError):
+            constraint = exception.constraint
+            syntax_problem = exception.parameter
+            highlighted_syntax = syntax_problem.replace(syntax_problem[exception.offset-1], "<b>%s</b>" % (syntax_problem[exception.offset-1]))
```

Comment:
If the error is at the letter 'a' for example, then this will bold-face every 'a'. If the error is at a blank space, then nothing will appear boldfaced.  Maybe the following, which highlights everything after the offset?
```python
highlighted_syntax = syntax_problem[:exception.offset]+"<b>"+syntax_problem[excpetion.offset:]+"</b>"
```